Training/Program.cs

```diff
@@ -1,7 +1,131 @@
+      public int Capacity => mVariables.Length;
+
+      /// <summary>The element at the specified index is get or set</summary>
+      /// <param name="index">Gets or sets the zero-based index of the element</param>
+      /// <returns>The element at the specified index</returns>
+      /// <exception cref="IndexOutOfRangeException">Thrown when the index is out of range [0, Count)</exception>
+      public T this[int index] {
+         get {
+            if (index < 0 || index >= mSize) throw new IndexOutOfRangeException ("Index is out of range");
+            return mVariables[index];
+         }
+         set {
+            if (index < 0 || index >= mSize) throw new IndexOutOfRangeException ("Index is out of range");
+            mVariables[index] = value;
+         }
+      }
+      #endregion
+
+      #region Methods -----------------------------------------------
+      /// <summary>Adds an element to the end of the list</summary>
+      /// <param name="element">The element to add to the list</param>
+      public void Add (T element) {
+         if (Count == Capacity) Array.Resize (ref mVariables, Capacity * 2);
+         mVariables[mSize++] = element;
+      }
+
+      /// <summary>Removes the first occurrence of a specific element from the list</summary>
+      /// <param name="element">The element to remove</param>
+      /// <returns>True if the element was successfully removed</returns>
+      public bool Remove (T element) {
+         if (mVariables.Contains (element)) {
+            mVariables = mVariables.Except (new T[] { element }).ToArray ();
+            return true;
+         } else throw new InvalidOperationException ("Item that is not found in the list");
+      }
+
+      /// <summary>Clears the entire list</summary>
+      public void Clear () => mSize = 0;
+
+      /// <summary>Insert elements at a specified index</summary>
+      /// <param name="index"></param>
+      /// <param name="element"></param>
+      /// <exception cref="IndexOutOfRangeException"></exception>
+      public void Insert (int index, T element) {
+         if (index < 0 || index >= mVariables.Length) throw new IndexOutOfRangeException ("Index is out of range");
+         if (Count == Capacity) Array.Resize (ref mVariables, Capacity * 2);
+         mVariables[mSize++] = element;
+         for (int i = mSize - 1; i >= index; i--) mVariables[i + 1] = mVariables[i];
+         mVariables[index] = element;
+      }
+
+      /// <summary>Removes the element at the specified index</summary>
+      /// <param name="index"></param>
+      /// <exception cref="IndexOutOfRangeException"></exception>
+      public void RemoveAt (int index) {
+         if (index < 0 || index >= mVariables.Length) throw new IndexOutOfRangeException ("Index is out of range");
+         for (int i = index; i < mVariables.Length - 1; i++)
+            mVariables[i] = mVariables[i + 1];
+         Array.Resize (ref mVariables, mVariables.Length - 1);

```

Comment:
What is the purpose of this?

Reply:
To shift the elements to left side, for example if 0th element is removed, then the 1st element must be shifted to 0th place.